docs/DeveloperGuide.adoc

```diff
@@ -192,6 +196,54 @@ image::add-ad.PNG[]
 ** Cons: Not as user-friendly.
 // end ::add[]
 
+// tag:stats[]
+=== Statistics feature : `stats`
+==== Implementation
+
+The stats feature allows the user to generate statistics about the eateries they have visited and reviews they have made at those visited eateries.
+
+Given below is an example usage scenario and how the stats mechanism behaves at each step.
+
+. The user launches the application for the first time. The `EateryList` will be initialised with the initial or previous json data stored.
+
+. The user executes `stats` to generate statistics for the current application state. The `EatMeParser` would return a `StatsCommand` to execute the command.
+
+. `StatsCommand` then creates `Statistics` to generate the different types of statistics needed. To create the different types of statistics, `Statistics` would, first, sort the eateries in `EateryList` based on different conditions i.e. categories, total expense, total number of visits. After sorting the eateries, `Statistics` would then use these sorted lists to generate the appropriate statistics.
+
+[NOTE]
+If the user executes the `stats` command in the Todo mode, `StatsCommand` would throw a `CommandException` to the user specifying that the user is in the wrong mode. In addition, if there is no eateries with any reviews in the application at the time of `stats` execution, the `StatsCommand` would throw a `CommandException` error saying that there is no data to generate statistics. Lastly, if there is any error during the generating of statistics, the `StatsCommand` will throw another `CommandException` saying that there is an unknown error and the application is unable to generate statistics.
+
+. `StatsCommand` would then call `Model#setStatistics()` with `Statistics` as a a parameter.
+
+. After successfully generating the statistics, `StatsCommand` will return the `CommandResult` to the `Ui`.
+
+The following sequence diagram shows how the statistics feature is executed.
+image:StatsSequenceDiagram.png[]
+
+Since the statistic feature has a different user interface compared to the other commands i.e. `AddCommand`, `AddTagCommand`, etc., it would be useful to explain how the different user interface components work.
+
+. When the `Ui` receives the `CommandResult` from `StatsCommand`, the `Ui` will call `MainWindow#showStats()`.
+
+. `MainWindow#showStats()` will then if `StatsWindow` is already open, if yes then it will call `StatsWindow#initStats()` and focuses the window. Else, it will call `StatsWindow#initStats()` and shows the window.
```

Comment:
`MainWindow#showStats()` will check if `StatsWindow` is already open. If it is opened, it will call `StatsWindow#initStats()` and focuses the window. Else, it will call `StatsWindow#initStats()` and displays the window.

btw what is focuses the window

Reply:
like... focuses the window.. bring it to the forefront ?